src/testRunner/unittests/tsc/projectReferences.ts

```diff
@@ -49,4 +49,49 @@ describe("unittests:: tsc:: projectReferences::", () => {
             }),
         commandLineArgs: ["--p", "src/project"],
     });
+
+    verifyTsc({
+        scenario: "projectReferences",
+        subScenario: "default import interop uses referenced project settings",
+        fs: () =>
+            loadProjectFromFiles({
+                "/node_modules/ambiguous-package/package.json": `{ "name": "ambiguous-package" }`,
+                "/node_modules/ambiguous-package/index.d.ts": "export declare const ambiguous: number;",
+                "/node_modules/esm-package/package.json": `{ "name": "esm-package", "type": "module" }`,
+                "/node_modules/esm-package/index.d.ts": "export declare const esm: number;",
+                "/lib/tsconfig.json": jsonToReadableText({
+                    compilerOptions: {
+                        composite: true,
+                        declaration: true,
+                        rootDir: "src",
+                        outDir: "dist",
+                        module: "esnext",
+                        moduleResolution: "bundler",
+                    },
+                    include: ["src"],
+                }),
+                "/lib/src/a.ts": "export const a = 0;",
+                "/lib/dist/a.d.ts": "export declare const a = 0;",
+                "/app/tsconfig.json": jsonToReadableText({
+                    compilerOptions: {
+                        module: "esnext",
+                        moduleResolution: "bundler",
+                        rootDir: "src",
+                        outDir: "dist",
+                    },
+                    include: ["src"],
+                    references: [
+                        { path: "../lib" },
+                    ],
+                }),
+                "/app/src/local.ts": "export const local = 0;",
+                "/app/src/index.ts": `
+                    import local from "./local"; // Error
+                    import esm from "esm-package"; // Error
+                    import referencedSource from "../../lib/src/a"; // Error
+                    import referencedDeclaration from "../../lib/dist/a"; // Error

```

Comment:
@sheetalkamat do you think it’s worth it to fix this? These two default imports would ideally error, because we could know by using the referenced project settings that the emit was ESM. We currently don’t error because we assume that a declaration file that says `export declare const a = 0` could represent a CJS file. I’m not sure if it should be out of scope for this PR to start using referenced project settings to get more accurate information during type checking like this.